config/backend-config/backend-config.go

```diff
@@ -288,9 +291,12 @@ WaitForConfig waits until backend config has been initialized
 */
 func (bc *CommonBackendConfig) WaitForConfig() {
 	for {
+		initializedLock.RLock()
```

Comment:
Same thing here. But it's a bit tricky as we are in an infinite for-loop. 
Thinking if we should have an anonymous function that does defer lock handling.

Eg.
```
for {
   func() {
      init.RLock()
      defer init.RUnlock()
      if initialized {
         break
      }
      ---
      ---
   }
}
```

Reply:
I think both are good. I am fine with either.

Reply:
Will break work here?

Reply:
Yeah, Chandu mentioned that it won't work. Resolving this.